packages/labs/src/components/select/multiSelect.tsx

```diff
@@ -0,0 +1,256 @@
+
+        return (
+            <Popover
+                autoFocus={false}
+                canEscapeKeyClose={true}
+                enforceFocus={false}
+                isOpen={this.state.isOpen}
+                position={Position.BOTTOM_LEFT}
+                {...popoverProps}
+                className={classNames(listProps.className, popoverProps.className)}
+                onInteraction={this.handlePopoverInteraction}
+                popoverClassName={classNames("pt-multi-select-popover", popoverProps.popoverClassName)}
+                popoverDidOpen={this.handlePopoverDidOpen}
+                popoverWillOpen={this.handlePopoverWillOpen}
+            >
+                <div
+                    onKeyDown={this.handleKeyDown}
+                    onKeyUp={this.state.isOpen && handleKeyUp}
+                >
+                    <TagInput
+                        inputProps={defaultInputProps}
+                        {...tagInputProps}
+                        className={classNames("pt-multi-select", tagInputProps.className)}
+                        values={this.props.selectedItems.map(this.props.tagRenderer)}
+                    />
+                </div>
+                <div onKeyDown={handleKeyDown} onKeyUp={handleKeyUp}>
+                    <ul className={Classes.MENU} ref={listProps.itemsParentRef}>
+                        {this.renderItems(listProps)}
+                    </ul>
+                </div>
+            </Popover>
+        );
+    }
+
+    private renderItems({ activeItem, filteredItems, handleItemSelect }: IQueryListRendererProps<T>) {
+        const { itemRenderer, noResults } = this.props;
+        if (filteredItems.length === 0) {
+            return noResults;
+        }
+        return filteredItems.map((item, index) => itemRenderer({
+            index,
+            item,
+            handleClick: (e) => handleItemSelect(item, e),
+            isActive: item === activeItem,
+        }));
+    }
+
+    private handleQueryChange = (event: React.FormEvent<HTMLInputElement>) => {
+        const query = event.currentTarget.value;
+        this.setState({ query, isOpen: !(query.length === 0 && this.props.openOnKeyDown) });
+    }
+
+    private handleItemSelect = (item: T, event: React.SyntheticEvent<HTMLElement>) => {
+        this.input.focus();
+        // make sure the query is valid by checking if activeItem is defined
+        if (this.state.activeItem) {
+            if (this.props.resetOnSelect && this.state.query.length > 0) {
+                this.setState({
+                    activeItem: this.props.items[0],
+                    query: "",
+                });
+            }
+            Utils.safeInvoke(this.props.onItemSelect, item, event);
+        }
+    }
+
+    private handlePopoverInteraction = (nextOpenState: boolean) => requestAnimationFrame(() => {
+        // deferring to rAF to get properly updated activeElement
+        const { popoverProps = {}, resetOnSelect } = this.props;
```

Comment:
again use `defaultProps` so that you're not setting the default in multiple places